packages/packages/store/src/__tests__/store.test.tsx

```diff
@@ -0,0 +1,271 @@
+		} );
+
+		// Assert.
+		expect( () => {
+			addSlice( {
+				name: 'slice',
+				initialState: {
+					value: 1,
+				},
+				reducers: {},
+			} );
+		} ).toThrow( 'Slice with name "slice" already exists.' );
+	} );
+
+	it( 'should throw an error when trying to re-create the store', () => {
+		// Arrange.
+		addSlice( {
+			name: 'slice',
+			initialState: {
+				value: 1,
+			},
+			reducers: {},
+		} );
+
+		createStore();
+
+		// Assert.
+		expect( () => createStore() ).toThrow( 'The store instance already exists.' );
+	} );
+
+	it( 'should add a middleware that blocks the state update by not running next(action)', () => {
+		// Arrange.
+		addMiddleware( () => () => () => {} );
+
+		const { slice, wrapper } = createStoreEntities();
+
+		// Act.
+		const { result } = renderHook( () => {
+			const dispatchAction = useDispatch();
+
+			dispatchAction( slice.actions.setValue( 4 ) );
+
+			return useSelector( ( state: SliceStateRoot ) => state.slice.value );
+		}, { wrapper } );
+
+		// Assert.
+		expect( result.current ).toBe( 1 );
+	} );
+
+	it( 'should add a middleware that does not interfere with the state update', () => {
+		// Arrange.
+		addMiddleware( () => ( next: Dispatch<AnyAction> ) => ( action: any ) => {
+			next( action );
+		} );
+
+		const { slice, wrapper } = createStoreEntities();
+
+		// Act.
+		const { result } = renderHook( () => {
+			const dispatchAction = useDispatch();
+
+			dispatchAction( slice.actions.setValue( 4 ) );
+
+			return useSelector( ( state: SliceStateRoot ) => state.slice.value );
+		}, { wrapper } );
+
+		// Assert.
+		expect( result.current ).toBe( 4 );
+	} );
+
+	it( 'should dispatch an action without using hooks', () => {
+		// Arrange.
+		addSlice( {
+			name: 'slice',
+			initialState: {
+				value: 1,
+			},
+			reducers: {
+				setValue: ( state, action ) => {
+					state.value = action.payload;
+				},
+			},
+		} );
+
+		const store = createStore();
+
+		// Act.
+		store.dispatch( { type: 'slice/setValue', payload: 6 } );
+
+		const stateResult = store.getState().slice.value;
+
+		// Assert.
+		expect( stateResult ).toBe( 6 );
+	} );
+
+	it( 'should collect actions that are dispatched before the store exist, and run them when the store instance is created', () => {
+		// Act.
+		dispatch( { type: 'slice/setValue', payload: 7 } );
+
+		const { wrapper } = createStoreEntities();
+
+		const { result } = renderHook( () => useSelector( ( state: SliceStateRoot ) => state.slice.value ), { wrapper } );
+
+		// Assert.
+		expect( result.current ).toBe( 7 );
+	} );
+
+	it( 'should delete the store instance', () => {
+		// Arrange.
+		const { store } = createStoreEntities();
+
+		let instance = getStore();
+
+		// Assert.
+		expect( instance ).toEqual( store );
+
+		// Act.
+		deleteStore();
+
+		instance = getStore();
+
+		// Assert.
+		expect( instance ).toBeNull();
+	} );
+
+	it( 'should delete the added slices', () => {
+		// Arrange.
+		jest.spyOn( console, 'error' ).mockImplementation( () => {} );
```

Comment:
add a comment about why we need this (maybe with references to the sources)

Reply:
The comment was on the assertion, moved it to the top.